AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/src/AwsCryptographyMaterialProvidersOperations.dfy

```diff
@@ -258,6 +259,26 @@ module AwsCryptographyMaterialProvidersOperations refines AbstractAwsCryptograph
   predicate CreateAwsKmsHierarchicalKeyringEnsuresPublicly(input: CreateAwsKmsHierarchicalKeyringInput, output: Result<IKeyring, Error>)
   {true}
 
+  function method N(n : PositiveLong) : string {
+    String.Base10Int2String(n as int)
+  }
+
+  method CheckCache(cache : CacheType, ttlSeconds: PositiveLong) returns (output : Result<bool, Error>)
```

Comment:
Given that this function is returning a bool, I think that it is better modeled as an `Outcome`?

Reply:
Whenever I use an Outcome, I eventually get bitten by the fact that Outcomes don't compose, and so I go back to Result. Now I just always start with Result.